### DISCUSSION TITLE
Experimental Addition, In Progress - A more "natural" Talk Feature

### DISCUSSION THREAD

**Discussion by sr229** [Monika-After-Story/MonikaModDev#7831] [general - submod / external content]
We've started this work around January 2018 and we're getting some progress from it. This will be merged to mainline MAS once deemed complete and feature-parity.

This issue tracker will serve as the Monika-After-Story's copy of our progress.

Expected Trajectory for release: Q2 2018.

### Current Working Group

- @sr229 (DevOps, Project Leader, ServOps)
- @fdzjuancarlos (Libitina TSC, Fumiko)
- @martmists (Libitina TSC, Fumiko)
- @Oxylibrium (Libitina TSC, Libitina)

### Repositories

- [ProjectMonika/Ivy](https://github.com/ProjectMonika/Libitina)

### TODOs

- [x] Finish up AI Component (Ivy)
- [ ] Finish up Dataset (Fern)
- [ ] Playtest-ready (Open Alpha, Open Beta)

**Comment by ThePotatoGuy** [comment]
Hey, you're still developing this as a sub-mod, correct?

**Comment by sr229** [comment]
@ThePotatoGuy yep! But this is planned to be merged to MAS Mainline as our original objective

**Comment by sr229** [comment]
Current problems atm:

- Once integrated. Monika will lose her visual emotion feedback per line. Investigating how we can circumvent this.

- Too high PC requirements. Probably only solution to support lower-end devices is using a ``online`` distribution (Headless, no AI server included, has to query a public server).

- We have limited lines. ~~and its hard porting all the MAS talk dialogue uwaa~~

**Comment by ThePotatoGuy** [comment]
I hope you're not porting _everything_. A chat ai is great for conversation-style dialogue, but not all of our dialogue is suited for that style.

**Comment by sr229** [comment]
@ThePotatoGuy well the joke here is even if you fit nonstandard dialogue to Monika, its still up to her to construct those

~~Based on our 2 months of intensive R&D and testing~~

**Comment by kaido1224** [comment]
So you were wanting to move everything to the chat bot AI?

**Comment by aldoram5** [comment]
Hi @sr229 !
I've been also working on a chat-talking way system for MAS for quite a long time too, that system is a retrieve based NLP powered one, mostly because I wanted to have full control over conversations to maintain the "Monika's voice" while also be able to work with the player input so Monika can "understand" and store player input for later. Also since it is totally custom and doesn't depend on third party libraries it plays nicely with renpy by default. 

I have my current legacy system here https://github.com/aldoram5/MonikaModDev/tree/legacy-chat-ai

I started rewriting some core parts of it and those are in another branch.

Since our goals are quite similar we could have a chat if you're interested.

**Comment by sr229** [comment]
@kaido1224 as far as we are concerened, only the talk feature

@aldoram5  sure! Give us a mail on monika@headbow.stream if you're interested merging with us for this project

**Comment by sr229** [comment]
@Swagsy the specs has been lowered down, so yes.

**Comment by ThePotatoGuy** [comment]
Just to clarify, @sr229 , your project is unrelated to the Project M(onika) thing by krazyjames, right?

The similar naming scheme is somewhat confusing.

**Comment by sr229** [comment]
@ThePotatoGuy we are not affiliated with that project, and this project is somewhat older as we started it at November 2017 under a internal codename "Evergreen"

**Comment by ThePotatoGuy** [comment]
Status of this has changed. closing unless project owners wish to reopen

**Comment by sr229** [comment]
Requesting Reopen: Project has resumed on a slower pace

**Comment by sr229** [comment]
Progress update: API is confirmed complete - working on documentation and starting Milestone 2 : Fern dataset.

**Comment by Storylover-Vodhr** [comment]
I'm looking forward to this one.

**Comment by Martmists-GH** [comment]
The project leader has been unable to work on the project for various reasons. As a result, development is on hiatus for undetermined amount of time.
